control.lua

```diff
@@ -156,165 +158,205 @@ end
-			factory.inside_energy_receiver.energy = 0
-			factory.outside_energy_sender.energy = 0
-			factory.outside_energy_receiver.energy = 0--e
-		else
-			factory.inside_energy_sender.energy = 0
-			factory.inside_energy_receiver.energy = 0--e
-			factory.outside_energy_sender.energy = 0--e
-			factory.outside_energy_receiver.energy = 0
+-- power poles can only connect to 5 other power poles. give priority to factory poles
+local our_poles = {["factory-power-connection"] = true, ["factory-power-pole"] = true, ["factory-overflow-pole"] = true}
+local function reduce_neighbours(pole)
+	local neighbours = pole.neighbours.copper
+	if #neighbours < 5 or #neighbours == 0 then return true end
+	
+	local n
+	for i, neighbour in ipairs(neighbours) do
+		if not our_poles[neighbour.name] and neighbour.surface == pole.surface then n = i break end
+	end
+	
+	if n == nil then
+		pole.surface.create_entity{name="flying-text", position=pole.position, text={"electric-pole-wire-limit-reached"}}
+		return false
+	end
+	
+	pole.disconnect_neighbour(neighbours[n])
+	return true
+end
+
+local function available_pole(factory)
+	local poles = factory.inside_power_poles
+	for i, pole in ipairs(poles) do
+		local next = poles[i+1]
+		if next then
+			next.connect_neighbour(pole)
 		end
 	end
-	if factory.energy_indicator and factory.energy_indicator.valid then
-		factory.energy_indicator.destroy()
-		factory.energy_indicator = nil
+	
+	for i, pole in ipairs(poles) do
+		if #pole.neighbours.copper < (i == #poles and 4 or 5) then return pole end
 	end
-	local direction = (factory.transfers_outside and defines.direction.south) or defines.direction.north
-	local energy_indicator = factory.inside_surface.create_entity{
-		name = "factory-connection-indicator-energy-d" .. make_valid_transfer_rate(factory.transfer_rate),
-		direction = direction, force = factory.force,
-		position = {x = factory.inside_x + factory.layout.energy_indicator_x, y = factory.inside_y + factory.layout.energy_indicator_y}
-	}
-	energy_indicator.destructible = false
-	factory.energy_indicator = energy_indicator
+	
+	local layout = factory.layout
+	local pole = factory.inside_surface.create_entity{name="factory-overflow-pole", position=poles[1].position, force=poles[1].force}
+	pole.destructible = false
+	pole.disconnect_neighbour()
+	pole.connect_neighbour(poles[#poles])
+	table.insert(poles, pole)
+	return pole
 end
--- For update 11
-function update_all_power_settings()
-	for _, factory in pairs(global.factories) do
-		update_power_settings(factory)
+
+local function connect_power(factory, pole)
+	if not reduce_neighbours(pole) then return end
+	factory.outside_power_pole = pole
+	
+	if factory.inside_surface.name ~= pole.surface.name then
+		available_pole(factory).connect_neighbour(pole)
+		factory.direct_connection = pole
+		return
 	end
+	
+	local n
+	for i, pole in ipairs(global.middleman_power_poles) do
+		if pole == 0 then n = i break end
+	end
+	n = n or #global.middleman_power_poles + 1
```

Comment:
This is also a continuation of the strange code pattern.

This would be a lot easier, cleaner, and readable, if you use the unit_number as the index, and loop using pairs